pyomo/core/base/set.py

```diff
@@ -621,7 +621,11 @@ def _get_discrete_interval(self):
         # but problemmatic for code coverage.
         ranges = list(self.ranges())
         if len(ranges) == 1:
-            start, end, c = ranges[0].normalize_bounds()
+            try:
+                start, end, c = ranges[0].normalize_bounds()
+            except AttributeError:
+                # Catching Any, NonNumericRange, etc...
+                return (None, None, None,)
```

Comment:
This is only necessary for the global set declaration, right?

Reply:
The new global set certainly exposed the bug, but it will show up with any set with a single non-numeric entry:

```python
>>> m = ConcreteModel()
>>> m.I = Set(initialize=['a'])
>>> m.I.get_interval()
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "pyomo/pyomo/core/base/set.py", line 611, in get_interval
    return self._get_discrete_interval()
  File "pyomo/pyomo/core/base/set.py", line 624, in _get_discrete_interval
    start, end, c = ranges[0].normalize_bounds()
AttributeError: 'NonNumericRange' object has no attribute 'normalize_bounds'
```

Reply:
This example pointed out a bug in the implementation.  That has been resolved (with new test cases covering the example)